tests/test_federal_senate_dataset.py

```diff
@@ -1,41 +1,92 @@
 
 from serenata_toolbox.federal_senate.federal_senate_dataset import FederalSenateDataset
 
+
 class TestFederalSenateDataset(TestCase):
-    def setUp(self):
+    @classmethod
+    def setUpClass(cls):
+        cls.expected_files = ['federal-senate-2008.csv',
+                              'federal-senate-2009.csv']
+
+    @patch('serenata_toolbox.federal_senate.federal_senate_dataset.urlretrieve')
+    def test_fetch_files_from_S3(self, mockedUrlRetrieve):
         self.path = gettempdir()
         self.subject = FederalSenateDataset(self.path)
 
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_fetch_saves_raw_files(self):
-        self.subject.fetch()
-        names = ['federal-senate-{}.csv'.format(year) for year in range(self.subject.FIRST_YEAR, self.subject.NEXT_YEAR)]
-        for name in names:
-            file_path = os.path.join(self.path, name)
-            assert(os.path.exists(file_path))
-
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_translate_creates_english_versions_for_every_csv(self):
-        self.subject.fetch()
-        self.subject.translate()
-        names = ['federal-senate-{}.xz'.format(year) for year in range(self.subject.FIRST_YEAR, self.subject.NEXT_YEAR)]
-        for name in names:
-            file_path = os.path.join(self.path, name)
-            assert(os.path.exists(file_path))
-
-    @skipIf(os.environ.get('RUN_INTEGRATION_TESTS') != '1',
-            'Skipping integration test')
-    def test_clean_creates_a_reimbursements_file(self):
-        self.subject.fetch()
+        retrieved_files, not_found_files = self.subject.fetch()
+
+        self.assertTrue(mockedUrlRetrieve.called)
+        self.assertEqual(mockedUrlRetrieve.call_count, 10)
+        for retrieved_file, expected_file in zip(
+                retrieved_files, self.expected_files):
+
+            self.assertIn(expected_file, retrieved_file)
+
+    def test_fetch_not_found_files_from_S3(self):
+        self.path = gettempdir()
+        self.subject = FederalSenateDataset(self.path, 2007, 2008)
+
+        retrieved_files, not_found_files = self.subject.fetch()
+
+        for not_found_file, expected_file in zip(
+                not_found_files, self.expected_files):
+
+            self.assertIn('federal-senate-2007.csv', not_found_file)
+
+    def test_dataset_translation(self):
+        self.subject = FederalSenateDataset(os.path.join('tests/fixtures/csv/'), 2008, 2009)
+
+        expected_files = ['federal-senate-2008.csv']
+
+        translated_files, not_found_files = self.subject.translate()
+
+        for translated_file, expected_file in zip(
+                translated_files, expected_files):
+
+            self.assertIn(expected_file, translated_file)
+
+    def test_if_translation_happened_as_expected(self):
+        self.subject = FederalSenateDataset(os.path.join('tests/fixtures/csv/'), 2008, 2009)
```

Comment:
Here the idea of using `os.path.join` is so that you do not have to set the path with the `/` separator like: `os.path.join('tests', 'fixtures', 'csv')`.

Why is that? Because the separator used in file paths on Windows is `\` and not `/` as Unix based systems ;)

So the output of `os.path.join('tests', 'fixtures', 'csv')` on:
- Windows would be `tests\fixtures\csv\`
- Unix based would be `tests/fixtures/csv/`

And this is only one of the reasons to use `os.path.join` =)